scripts/install/setup.sh

```diff
@@ -1,21 +1,75 @@
 #!/usr/bin/env bash
 
-bold() {
-  echo ". $(tput bold)" "$*" "$(tput sgr0)";
-}
-
 err() {
   echo "$*" >&2;
 }
 
-~/spinnaker-for-gcp/scripts/manage/check_duplicate_dirs.sh || exit 1
-~/spinnaker-for-gcp/scripts/manage/check_git_config.sh || exit 1
+[ -z "$REPO_PATH" ] && REPO_PATH="$HOME"
+
+source $REPO_PATH/spinnaker-for-gcp/scripts/manage/service_utils.sh
+
+REQUIRED_BINARIES=(git gcloud jq kubectl)
+
+MISSING_BINARIES=""
+for b in "${REQUIRED_BINARIES[@]}"; do
+  BINARY_PATH=$(check_for_command $b)
+  if [ -z "$BINARY_PATH" ]; then
+    if [ -z $MISSING_BINARIES ]; then
+      MISSING_BINARIES="$b"
+    else 
+      MISSING_BINARIES="$MISSING_BINARIES, $b"
+    fi
+  fi
+done
+
+if [ -n "$MISSING_BINARIES" ]; then 
+  bold "The following command(s) are required for setup but were not found: $MISSING_BINARIES"
+  exit 1
+fi
 
-PROPERTIES_FILE="$HOME/spinnaker-for-gcp/scripts/install/properties"
+REPO_PATH=$REPO_PATH $REPO_PATH/spinnaker-for-gcp/scripts/manage/check_git_config.sh || exit 1
+
+[ -z "$PROPERTIES_FILE" ] && PROPERTIES_FILE="$REPO_PATH/spinnaker-for-gcp/scripts/install/properties"
 
 source "$PROPERTIES_FILE"
 
-~/spinnaker-for-gcp/scripts/manage/check_project_mismatch.sh
+if [ "$PROJECT_ID" != "$NETWORK_PROJECT" -a "$CI" = true ]; then
```

Comment:
Same comment here about pulling this routine out into a function. Want to try to keep the size of the `setup.sh` file reasonable.